pkg/util/algorithms/math.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2023 ScyllaDB.
+
+package algorithms
+
+import "github.com/scylladb/scylla-operator/pkg/util/constraints"
+
+func Max[T constraints.Number](a T, tail ...T) T {
```

Comment:
Why not use `Ordered` as a constraint here?

Reply:
Max on string kinda doesn't makes sense

Reply:
It wouldn't be ordered otherwise. And I believe wanting to find a lexicographically greatest/lowest string is common enough.

Reply:
seems like other languages are following this: https://en.cppreference.com/w/cpp/algorithm/max
i'll make it consistent